presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -382,6 +382,12 @@ public PlanWithProperties visitExchange(ExchangeNode node, StreamPreferredProper
         {
             checkArgument(node.getScope() != LOCAL, "AddLocalExchanges can not process a plan containing a local exchange");
             // this node changes the input organization completely, so we do not pass through parent preferences
+            if (node.getOrderingScheme().isPresent()) {
+                return planAndEnforceChildren(
+                        node,
+                        any().withOrderSensitivity(),
+                        any().withOrderSensitivity());
+            }
```

Comment:
How is order sensitivity being enforced? My reading of `AddLocalExchanges.Rewriter#enforce` suggests that if plan doesn't satisfy order sensitivity already, the data will get partitioned using FIXED_ARBITRARY_DISTRIBUTION.

Furthermore, `StreamPreferredProperties#isSatisfiedBy` implementation suggests that `any().withOrderSensitivity()` will be satisfied by any plan. This doesn't seem right.

```
    public boolean isSatisfiedBy(StreamProperties actualProperties)
    {
        // is there a specific preference
        if (!distribution.isPresent() && !partitioningColumns.isPresent()) {
            return true;
        }
```